### PR TITLE
refactor (forms): using more utility functions and optimizing is-true condition statements

### PR DESCRIPTION
**Pull request by vyom1611**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/forms/src/validators.ts [hunk 1]
```diff
@@ -483,7 +483,7 @@ export function requiredValidator(control: AbstractControl): ValidationErrors|nu
  * See `Validators.requiredTrue` for additional information.
  */
 export function requiredTrueValidator(control: AbstractControl): ValidationErrors|null {
-  return control.value === true ? null : {'required': true};
+  return (control.value) ? null : {'required': true};
```
Comment (dylhunn) [inline packages/forms/src/validators.ts:486]: I don't think this is right. What if `control.value` is `null`?

File: packages/forms/src/validators.ts [hunk 2]
```diff
@@ -630,7 +630,7 @@ export function normalizeValidators<V>(validators: (V|Validator|AsyncValidator)[
 function compose(validators: (ValidatorFn|null|undefined)[]|null): ValidatorFn|null {
   if (!validators) return null;
   const presentValidators: ValidatorFn[] = validators.filter(isPresent) as any;
-  if (presentValidators.length == 0) return null;
+  if (!hasValidLength(presentValidators)) return null;
```
Comment (dylhunn) [inline packages/forms/src/validators.ts:633]: This appears to be an entirely different check:

```
return value != null && typeof value.length === 'number';
```

Is this a safe substitution? It doesn't seem so to me.

**Review by dylhunn — CHANGES_REQUESTED** [review]

**Comment by dylhunn** [comment]
P.S. I think as a matter of code style, we generally prefer to keep explicit checks rather than relying on implicit comparisons and coercion.

**Comment by vyom1611** [comment]
Why does the validLength function not work on line 633, the original condition was checking for length to be 0, which would be falsy

**Comment by AndrewKushnir** [comment]
Closing this PR, since it changes the code to rely on implicit comparisons and coercion, which we generally try to avoid.

**Pull request closed without merging**